enhancements/kubelet/virtualization-higher-workload-density.md

```diff
@@ -0,0 +1,570 @@
+Add e2e tests for the WASP agent repository for regression testing against
+OpenShift.
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+
+Dev Preview is not planned.
+
+### Tech Preview -> GA
+
+Specific needs
+- SWAP triggered eviction for workloads
+
+Generic needs
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+#### Swap GA for CNV users only
+In addition, swap is intended to become GA only for CNV users.
+
+This will be achieved in the following way:
+- The system will figure out if CNV is enabled by checking if the `hyperconvergeds.hco.kubevirt.io`
+object and the `openshift-cnv` namespace exist.
+- If CNV is enabled, do not emit an alert.
+- Otherwise, emit an alert that says this cluster has swap enabled and it is not supported.
+- Guard against non CNV users setting `LimitedSwap` in the KubeletConfig via MCO validation logic.
+
+### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+## Upgrade / Downgrade Strategy
+
+If applicable, how will the component be upgraded and downgraded? Make sure this
+is in the test plan.
+
+Consider the following in developing an upgrade/downgrade strategy for this
+enhancement:
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to keep previous behavior?
+- What changes (in invocations, configurations, API use, etc.) is an existing
+  cluster required to make on upgrade in order to make use of the enhancement?
+
+Upgrade expectations:
+- Each component should remain available for user requests and
+  workloads during upgrades. Ensure the components leverage best practices in handling [voluntary
+  disruption](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/). Any exception to
+  this should be identified and discussed here.
+- Micro version upgrades - users should be able to skip forward versions within a
+  minor release stream without being required to pass through intermediate
+  versions - i.e. `x.y.N->x.y.N+2` should work without requiring `x.y.N->x.y.N+1`
+  as an intermediate step.
+- Minor version upgrades - you only need to support `x.N->x.N+1` upgrade
+  steps. So, for example, it is acceptable to require a user running 4.3 to
+  upgrade to 4.5 with a `4.3->4.4` step followed by a `4.4->4.5` step.
+- While an upgrade is in progress, new component versions should
+  continue to operate correctly in concert with older component
+  versions (aka "version skew"). For example, if a node is down, and
+  an operator is rolling out a daemonset, the old and new daemonset
+  pods must continue to work correctly even while the cluster remains
+  in this partially upgraded state for some time.
```

Comment:
this seems like generic content. should we not replace it with something specific, or drop it?